### PR TITLE
import statistics

### PR DESCRIPTION
**Pull request by bend-n**
image

imported statistics (also chain)

### CODE REVIEW

File: bot.py [hunk 1]
```diff
@@ -236,25 +236,29 @@ async def benchmark(msg: discord.Message, code: bytes, day: int, part: int) -> N
...
-        await msg.reply(embed=discord.Embed(title="Benchmark complete", description=f"Median: **{ns(median)}**\nAverage: **{ns(average)}**"))
-    else:
-        #await msg.reply(embed=discord.Embed(title="Benchmark complete (Unverified)", description=f"Median: **{ns(median)}**\nAverage: **{ns(average)}**\nTotal Memory Accesses: **{total_memory_accesses:,.2f}**\nTotal L1 I-Cache Misses: **{total_l1_icache_misses:,.2f}**\nTotal LL I-Cache Misses: **{total_ll_icache_misses:,.2f}**\nTotal L1 D-Cache Misses: **{total_l1_dcache_misses:,.2f}**\nTotal LL D-Cache Misses: **{total_ll_dcache_misses:,.2f}**"))
-        await msg.reply(embed=discord.Embed(title="Benchmark complete (Unverified)", description=f"Median: **{ns(median)}**\nAverage: **{ns(average)}**"))
+    best = min([int(r["median"]) for r in results])
```
Comment (indiv0) [inline bot.py:239]: Should best be `min(median)`? Not `min(min)`?

File: bot.py [hunk 2]
```diff
@@ -236,25 +236,29 @@ async def benchmark(msg: discord.Message, code: bytes, day: int, part: int) -> N
...
+    med = median([int(r["median"]) for r in results])
+    dev = stdev(chain([int(r["min"]) for r in results], [int(r["max"]) for r in results]))
+    #total_memory_accesses = mean([int(r["total_memory_accesses"]) for r in results])
+    #total_l1_icache_misses = mean([int(r["total_l1_icache_misses"]) for r in results])
+    #total_ll_icache_misses = mean([int(r["total_ll_icache_misses"]) for r in results])
+    #total_l1_dcache_misses = mean([int(r["total_l1_dcache_misses"]) for r in results])
+    #total_ll_dcache_misses = mean([int(r["total_ll_dcache_misses"]) for r in results])
+
+    title = "Benchmark complete" if verified else "Benchmark complete (Unverified)"
+    negligible = (abs(previous_best - best) < 200) if best > 1000 else (abs(previous_best - best) < 50)
```
Comment (indiv0) [inline bot.py:249]: What if `previous_best` is `None`?

File: bot.py (hunk 1 above)
Comment (bend-n) [reply bot.py:239]: Thats what the leaderboard uses, so it would be strange to have two different numbers?

File: bot.py (hunk 2 above)
Comment (bend-n) [reply bot.py]: good catch, handled

File: bot.py (hunk 1 above)
Comment (indiv0) [reply bot.py:241]: fair enough

**Comment by indiv0** [comment]
LGTM, thanks!

**Pull request merged**